# Data selection for comparative genomic analysis of vertebrate _L. johnsonii_ strains
At the time of the analysis (04/17/23), there were 158 _L. johnsonii_ assemblies  in the NCBI database. 
1. Entrez Direct was used to pull down all the assembly accession numbers.

```
esearch -db assembly -query 'Lactobacillus johnsonii'| efetch -format docsum \
| xtract -pattern DocumentSummary \
-element AssemblyAccession,BioSampleAccn,Sub_value,AssemblyStatus,FtpPath_RefSeq > assembly_query_output.txt
```

2. From the list of assembly accession numbers extracted, those related to _l. johnsonii_ whole genome sequence were selected. Further attribute information on these assemblies was extracted using the Biosample accession number.

```
for acc in `cat biosampleAccesson_list.txt`;
do 
esearch -db assembly -query $acc| elink -target biosample | efetch -format docsum \
| xtract -pattern DocumentSummary -element Accession -block Attribute -pfx "\t|" -sep "|" -tab "" \
-element Attribute@display_name,Attribute >> biosampleAccesson_list_output.txt;
## this gives every single attribute that is available for the biosample, the key name and value separated by  "|" and each pair separated by '\t'.
done
sed -n 's/|strain|//g' -n 's/|isolation source|//g' -n 's/|host|//g' -n 's/|geographic location|//g' biosampleAccesson_list_output.txt
```

3. The next filtering was to remove assemblies that did not have a RefSeq fttp link. This resulted in a count of 85 genome assemblies.
4. Of the 85 assemblies, 56 genome assemblies of strains were isolated from vertebrate gastrointestinal tracts.
5. To avoid sampling bias, multiple genome assemblies that were submitted under the same BioProject, were filtered, and selected for a representative assembly. The selection was done based on the following thresholds:
	- Pairwise whole genome sequence average nucleotide identity (ANI) score of the assemblies is <0.999.
	- Has the least number of sequences contigs in the group.
	- Has the largest relative length of the longest and shortest contig.
9. Finally, if the number of genome assemblies in each host group was less than 5, they were removed from the analysis. This resulted in a final data size of 42 genome assemblies.
    
| Host of origin  | Number of assemblies |
| ------------- | ------------- |
| Rodent  | 17  |
| Human  | 10  |
| Avian | 9 |
| Swine | 6 |

# Downloading the selected genome assemblies
The RefSeq assemblies of the 42 strains were downloaded by creating a space-separated query text file with the assembly accession number and strain name.

```
cat query.txt |while read -r g
do 
query=$(echo $g | awk 'BEGIN{FS=" "}{print $1}')
esearch -db assembly -query $query </dev/null \
        | esummary \
        | xtract -pattern DocumentSummary -element FtpPath_RefSeq \
        | while read -r url ; do
            fname=$(echo $url | grep -o 'GCF_.*' | sed 's/$/_genomic.fna.gz/') ;
            wget "$url/$fname" ;
        done ;
done
```

