# Identifying single-copy core gene clusters
The maximum likelihood tree was built using the alignment of single-copy core gene clusters. The single-copy core gene clusters were first identified using the anvi'o pangenome interface by setting the following thresholds
Rule | Value
--- | --- 
Min number of genomes gene cluster occurs | 'num of genomes in the pangenome'
Max number of genes from each genome | 1
Min geometric homogeneity index | 1

The identified gene clusters were then saved under a bin labeled SCG. Once the collection bin of all the target gene clusters was set up, the amino acid sequences were extracted and concatenated.

```
anvi-get-sequences-for-gene-clusters -p LJ_Gut/LJ_Gut-PAN.db \
                                       -g LJ-GENOMES.db \
                                       -C SCG -b SCG \
                                       --concatenate-gene-clusters \
                                       -o SCG_Gut.fa
```
# Maximum liklihood analysis
A robust phylogenetic tree was then built using the SCG_gut.fa file, and two packages, trimal and iqtree.

```
# Trim the fasta file
# Removing sequences that had over 50% gap sequences
trimal -in SCG_Gut.fa -out SCG_Gut-clean.fa -gt 0.50

# Run phylogenetic analysis using IQtree
iqtree -s SCG_Gut-clean.fa -nt 8 -m WAG -bb 1000
```

The consensus tree file generated was then visualized using Phylo.io
